src/NUnitTestAdapter/TestConverter.cs

```diff
@@ -199,6 +199,35 @@ private VSTestResult GetBasicResult(XmlNode resultNode)
             if (outputNode != null)
                 vsResult.Messages.Add(new TestResultMessage(TestResultMessage.StandardOutCategory, outputNode.InnerText));
 
+            var attachments = resultNode.SelectNodes("attachments/attachment");
+            if (attachments.Count > 0)
+            {
+                var attachmentSet = new AttachmentSet(new Uri(NUnitTestAdapter.ExecutorUri), "Attachments");
+                vsResult.Attachments.Add(attachmentSet);
+
+                foreach (XmlNode attachment in attachments)
+                {
+                    var path = attachment.SelectSingleNode("filePath")?.InnerText ?? string.Empty;
+                    var description = attachment.SelectSingleNode("description")?.InnerText;
+
+                    try
+                    {
+                        // We only support absolute paths since we dont lookup working directory here
+                        // any problem with path will throw an exception
+                        var fileUri = new Uri(path, UriKind.Absolute);
+                        attachmentSet.Attachments.Add(new UriDataAttachment(fileUri, description));
+                    }
+                    catch (UriFormatException ex)
+                    {
+                        _logger.Warning(string.Format("Ignoring attachment with path '{0}' due to problem with path: {1}", path, ex.Message));
```

Comment:
Those are all reasons why I should not have stopped working on the projects accoding to you - and yet I did.

Some people prefer to use some things. You want to prioritize those people who prefer the "new thing." I want to take the conservative approach. This may be a basic personality disorder on the part of one of us. 😄 

When I was a very young developer in my 50s I thought that those folks who used the latest stuff were somehow better than those who don't. It's a typical attitude and very like the attitude at some large software companies. The message from NUnit has always been that we value users and contributors even where circumstances force them to use older releases of .NET, of VS or some open source IDE. I already (in our governance discussions) predicted this would change and I'm seeing it happen. It may be endemic to the .NET world itself.

If all that stuff were really available as Open Source __now__ I might see it differently. Anyway, it's not something I need to have a future say in.